services/weather/api.go

```diff
@@ -63,6 +63,7 @@ func (w *weatherService) fetchOWCurrent(lat, lon float64) (owCurrentResponse, er
 		return owCurrent, fmt.Errorf("executing http request: %w", err)
 	}
 
+	defer resp.Body.Close()
```

Comment:
Reviewer pointed out that I forgot to close the body after reading! 

Need to do a little refresher on how exactly those io.Readers and Writers work under the hood.